src/main/java/nextstep/subway/path/domain/CustomWeightedEdge.java

```diff
@@ -0,0 +1,26 @@
+package nextstep.subway.path.domain;
+
+import lombok.Getter;
+import nextstep.subway.path.application.dto.PathSearchType;
+import org.jgrapht.graph.DefaultWeightedEdge;
+
+@Getter
+public class CustomWeightedEdge extends DefaultWeightedEdge {
+    private PathSearchType type;
+    private Long distance;
+    private Long duration;
+
+    public CustomWeightedEdge(PathSearchType type, Long distance, Long duration) {
+        this.type = type;
+        this.distance = distance;
+        this.duration = duration;
+    }
+
+    @Override
+    protected double getWeight() {
+        if (type == PathSearchType.DURATION) {
```

Comment:
값을 비교하는 대신에 메시지를 보내는 방식으로 변경해보면 어떨까요?
```suggestion
        if (type.isDistance()) {
```